fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergeTwoAggRule.java

```diff
@@ -0,0 +1,132 @@
+//          |
+//      Aggregate
+//          |
+//         Node
+//
+// After:
+//      Aggregate
+//          |
+//         Node
+public class MergeTwoAggRule extends TransformationRule {
+    private static final List<String> ALLOW_MERGE_AGGREGATE_FUNCTIONS = Lists.newArrayList(FunctionSet.SUM,
+            FunctionSet.MAX, FunctionSet.MIN, FunctionSet.BITMAP_UNION, FunctionSet.HLL_UNION,
+            FunctionSet.PERCENTILE_UNION);
+
+    public MergeTwoAggRule() {
+        super(RuleType.TF_MERGE_TWO_AGG_RULE, Pattern.create(OperatorType.LOGICAL_AGGR).addChildren(
+                Pattern.create(OperatorType.LOGICAL_PROJECT)
+                        .addChildren(Pattern.create(OperatorType.LOGICAL_AGGR, OperatorType.PATTERN_LEAF))));
+    }
+
+    @Override
+    public boolean check(OptExpression input, OptimizerContext context) {
+        LogicalAggregationOperator aggregate1 = (LogicalAggregationOperator) input.getOp();
+        LogicalProjectOperator project = (LogicalProjectOperator) input.getInputs().get(0).getOp();
+        LogicalAggregationOperator aggregate2 =
+                (LogicalAggregationOperator) input.getInputs().get(0).getInputs().get(0).getOp();
+
+        // Only handle column prune, forbidden expression
+        if (!project.getColumnRefMap().entrySet().stream().allMatch(e -> e.getKey().equals(e.getValue()))) {
+            return false;
+        }
+
+        if (aggregate2.getPredicate() != null || aggregate2.hasLimit()) {
+            return false;
+        }
+
+        // The data distribution of the aggregate nodes keeps same
+        if (!aggregate2.getGroupingKeys().containsAll(aggregate1.getGroupingKeys())) {
+            return false;
+        }
+
+        Map<ColumnRefOperator, CallOperator> aggCallMap1 = aggregate1.getAggregations();
+        Map<ColumnRefOperator, CallOperator> aggCallMap2 = aggregate2.getAggregations();
+
+        if (!aggCallMap1.values().stream().allMatch(
+                c -> c.isAggregate() && c.getUsedColumns().cardinality() == 1 && c.getChild(0).isColumnRef())) {
+            return false;
+        }
+
+        for (CallOperator call : aggCallMap1.values()) {
+            ColumnRefOperator ref = (ColumnRefOperator) call.getChild(0);
+            if (!ALLOW_MERGE_AGGREGATE_FUNCTIONS.contains(call.getFnName().toLowerCase())) {
+                return false;
+            }
+
+            if (!aggCallMap2.containsKey(ref) && !FunctionSet.MAX.equalsIgnoreCase(call.getFnName()) &&
+                    !FunctionSet.MIN.equalsIgnoreCase(call.getFnName())) {
+                return false;
+            }
+
+            if (!aggCallMap2.get(ref).getFnName().equalsIgnoreCase(call.getFnName())) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    @Override
+    public List<OptExpression> transform(OptExpression input, OptimizerContext context) {
+        LogicalAggregationOperator aggregate1 = (LogicalAggregationOperator) input.getOp();
```

Comment:
`aggregateAbove` and `aggregateBelow`  is  better than `aggregate1` and `aggregate2`